kernel/kernel-impl/src/main/java/org/sakaiproject/authz/impl/SakaiSecurity.java

```diff
@@ -705,6 +724,52 @@ protected boolean checkAuthzGroups(String userId, String function, String entity
 			azgs = ref.getAuthzGroups(userId);
 		}
 
+		// need to know whether role swap is in effect, since we can't share the cache entry between sessions
+		// that are swapped and not swapped
+
+		String siteRef = null;
+		String roleswap = null;
+
+		// Actual code in DbAuthzGroupService will not roleswap if there's a user site ref in the list and
+		// it is acceptable. However we can't tell that without doing a database access, so be conservative
+		// and cache it separately as a role swap
+		// This code does not handle delegated access
+		if (azgs != null && userId != null && userId.equals(sessionManager().getCurrentSessionUserId())) {
```

Comment:
I’m not opposed to that, but some of the changes in DbAuthzService can’t be done in SakaiSecurity. E.g. they need to check for a user realm, which we can’t do without a DB query. They need to check for delegated access, which I think we don’t need to do (because even with delegated access, rights to a specific resource by a specific user is always the same.) The code in DBAuthzService also sets up Sql statements that we don’t need.

You could use the same code for both, but it would need to have an argument specifying which context it is in, and I think the resulting code would be more complex than just putting a simplified version in SakaiSecurity

On May 18, 2016, at 9:44:44 AM, Earle Nietzel <notifications@github.com<mailto:notifications@github.com>> wrote:


In kernel/kernel-impl/src/main/java/org/sakaiproject/authz/impl/SakaiSecurity.java<https://github.com/sakaiproject/sakai/pull/2585#discussion_r63704352>:

> @@ -705,6 +724,52 @@ protected boolean checkAuthzGroups(String userId, String function, String entity
>                       azgs = ref.getAuthzGroups(userId);
>               }
>
> +             // need to know whether role swap is in effect, since we can't share the cache entry between sessions
> +             // that are swapped and not swapped
> +
> +             String siteRef = null;
> +             String roleswap = null;
> +
> +             // Actual code in DbAuthzGroupService will not roleswap if there's a user site ref in the list and
> +             // it is acceptable. However we can't tell that without doing a database access, so be conservative
> +             // and cache it separately as a role swap
> +             // This code does not handle delegated access
> +             if (azgs != null && userId != null && userId.equals(sessionManager().getCurrentSessionUserId())) {


Does it make sense to put this code in DbAuthzGroupService and then just call it from SecurityService to prevent duplicate code?

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/sakaiproject/sakai/pull/2585/files/9a50b3c1fd20204443977f1a856c9e1ae3809003#r63704352>